app/api_v1/builds.py

```diff
@@ -0,0 +1,313 @@
+
+       {
+           "bucket_name": "an-s3-bucket",
+           "bucket_root_dir": "lsst_apps/builds/b1",
+           "date_created": "2016-03-01T10:21:27.583795Z",
+           "date_ended": null,
+           "git_refs": [
+               "master"
+           ],
+           "github_requester": "jonathansick",
+           "product_url": "http://localhost:5000/products/lsst_apps",
+           "self_url": "http://localhost:5000/builds/1",
+           "slug": "b1",
+           "uploaded": false
+       }
+
+    :reqheader Authorization: Include the token in a username field with a
+        blank password; ``<token>:``.
+    :param slug: Product slug.
+
+    :<json array git_refs: Git ref array that describes the version of the
+        documentation being built. Typically this array will be a single
+        string, e.g. ``['master']`` but may be a list of several refs for
+        multi-package builds with ltd-mason.
+    :<json github_requester: Optional GitHub username handle of person
+        who triggered the build.
+    :<json string slug: Optional URL-safe slug for the build. If a slug is
+        not specified, then one will automatically be specified.
+
+    :>json string bucket_name: Name of the S3 bucket hosting the built
+        documentation.
+    :>json string bucket_root_dir: Directory (path prefix) in the S3 bucket
+        where this documentation build is located.
+    :>json string date_created: UTC date time when the build was created.
+    :>json string date_ended: UTC date time when the build was deprecated;
+        will be ``null`` for builds that are *not deprecated*.
+    :>json array git_refs: Git ref array that describes the version of the
+        documentation being built. Typically this array will be a single
+        string, e.g. ``['master']`` but may be a list of several refs for
+        multi-package builds with ltd-mason.
+    :>json string github_requester: GitHub username handle of person
+        who triggered the build (null is not available).
+    :>json string product_url: URL of parent product entity.
+    :>json string self_url: URL of this build entity.
+    :>json string slug: Slug of build; URL-safe slug. Will be unique to the
+        Product.
+    :>json string uploaded: True if the built documentation has been uploaded
+        to the S3 bucket. Use :http:post:`/builds/(int:id)/uploaded` to
+        set this to `True`.
+
+    :resheader Location: URL of the created build.
+
+    :statuscode 201: No error.
+    :statuscode 404: Product not found.
+    """
+    product = Product.query.filter_by(slug=slug).first_or_404()
+    build = Build(product=product)
+    try:
+        build.import_data(request.json)
+        db.session.add(build)
+        db.session.commit()
+    except Exception:
+        db.session.rollback()
+        raise
+    return jsonify(build.export_data()), 201, {'Location': build.get_url()}
+
+
+@api.route('/builds/<int:id>/uploaded', methods=['POST'])
```

Comment:
This could be a PATCH on a build instead of a POST. Usually a POST implies creation or some sort of side effect. I.e. If it were actually uploading the build then that would be a POST.